### PR TITLE
Simplify code only configuration

### PR DESCRIPTION
**Pull request by SeanFeldman**
Connects to #92

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@Particular/azure-maintainers this is ready for a review. Still marked as a `[wip]` because of the [last item](https://github.com/Particular/NServiceBus.AzureServiceBus/issues/92#issuecomment-193514070) on the list I'm not entirely sure we should go with.

File: src/AcceptanceTests/TransportEncoding/When_receiving_a_message_with_unknown_transport_encoding.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public Sender()
                 EndpointSetup<DefaultServer>(busConfiguration =>
                 {
                     // TODO: this is wrong. We should be able to configure serialization w/o going through topology
```
Comment (yvesgoeleven) [inline src/AcceptanceTests/TransportEncoding/When_receiving_a_message_with_unknown_transport_encoding.cs]: this comment can go now

File: src/Tests/API/APIApprovals.ApproveAzureServiceBusTransport.approved.txt [hunk 2]
```diff
@@ -719,4 +675,13 @@ namespace NServiceBus
         public NServiceBus.AzureServiceBusValidationSettings UseSubscriptionPathMaximumLength(int subscriptionPathMaximumLength) { }
         public NServiceBus.AzureServiceBusValidationSettings UseTopicPathMaximumLength(int topicPathMaximumLength) { }
     }
+    public class SizeInMegabytes
+    {
+        public static NServiceBus.SizeInMegabytes Size1024;
```
Comment (yvesgoeleven) [inline src/Tests/API/APIApprovals.ApproveAzureServiceBusTransport.approved.txt]: aren't these supposed to be initialized immediately?

File: src/Tests/Receiving/When_comparing_performance_for_prefetch.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
+namespace NServiceBus.Azure.WindowsAzureServiceBus.Tests.Sending
```
Comment (yvesgoeleven) [inline src/Tests/Receiving/When_comparing_performance_for_prefetch.cs]: wrong namesapce

File: src/Tests/API/APIApprovals.ApproveAzureServiceBusTransport.approved.txt (hunk 2 above)
Comment (SeanFeldman) [inline src/Tests/API/APIApprovals.ApproveAzureServiceBusTransport.approved.txt]: Strange, it is [here](https://github.com/Particular/NServiceBus.AzureServiceBus/blob/9628815e5ea83abf6e3a7ea89aa3126ffad49157/src/Transport/Config/ExtensionPoints/SizeInMegabytes.cs#L10)

File: src/Tests/Receiving/When_comparing_performance_for_prefetch.cs (hunk 3 above)
Comment (SeanFeldman) [inline src/Tests/Receiving/When_comparing_performance_for_prefetch.cs]: Moved

**Comment by SeanFeldman** [comment]
@Particular/azure-maintainers ready to be merged

**Pull request merged**